Makefile

```diff
@@ -46,10 +46,10 @@ GCFLAGS?=
 UPLOAD_CMD=$(KOPS_ROOT)/hack/upload ${UPLOAD_ARGS}
 
 # Unexport environment variables that can affect tests and are not used in builds
-unexport AWS_ACCESS_KEY_ID AWS_REGION AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN CNI_VERSION_URL DNS_IGNORE_NS_CHECK DNSCONTROLLER_IMAGE DO_ACCESS_TOKEN GOOGLE_APPLICATION_CREDENTIALS
-unexport KOPS_BASE_URL KOPS_CLUSTER_NAME KOPS_RUN_OBSOLETE_VERSION KOPS_STATE_STORE KOPS_STATE_S3_ACL KUBE_API_VERSIONS NODEUP_URL OPENSTACK_CREDENTIAL_FILE SKIP_PACKAGE_UPDATE
-unexport SKIP_REGION_CHECK S3_ACCESS_KEY_ID S3_ENDPOINT S3_REGION S3_SECRET_ACCESS_KEY
-
+unexport AWS_ACCESS_KEY_ID AWS_REGION AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN CNI_VERSION_URL DNS_IGNORE_NS_CHECK DNSCONTROLLER_IMAGE GOOGLE_APPLICATION_CREDENTIALS
```

Comment:
@rifelpet - thanks a lot for all the help. The tests are now passing.
Wanted your thoughts on this change - are we okay to remove the unexport option on some of these environment variables?

Reply:
I'm debating whether we should keep the `kubetest2 kops` invocations in e2e.mk at all. I originally added it there so I could iterate on it as a presubmit quickly, but the real presubmit and periodic jobs dont rely on the kubetest2 command in the makefile at all, they invoke `kubetest2 kops` directly in the prow job yaml. I think the DO periodic and/or presubmit jobs should do the same.

I do think theres value in having the DO kubetest2 command already setup somewhere, perhaps we can replace the command in e2e.mk with a shell script like this [upgrade test scenario](https://github.com/kubernetes/kops/blob/master/tests/e2e/scenarios/upgrade/run-test)? Create a new `tests/e2e/scenarios/digital-ocean/run-test` script and revert the changes in Makefile and e2e.mk.

If we do that then the next step would be to add the new periodic job and update the presubmit job to invoke kubetest2 directly.

Reply:
This sounds good, thanks for the direction @rifelpet .

Reply:
@rifelpet - Sorry for the delay - I have made the changes. Please have a look when you get a chance. Another question I had was invoking these scripts? So the `tests/e2e/scenarios/digital-ocean/run-test` is only for reference and not invoked from the real presubmit job correct?

Reply:
I agree that we could use this script as reference but have the presubmit job invoke similar kubetest2 commands directly. We'll need to update the presubmit job in test-infra to reflect these changes.

One more thing is that we just merged https://github.com/kubernetes/kops/pull/10847 which adds a --provider flag when calling the e2e.test binary. From searching the kubernetes/kubernetes/test/e2e directory, I don't see any digitalocean code including any [DO provider registered with the e2e tests](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/framework/providers). I think we'll run into [this "unknown provider"](https://github.com/kubernetes/kubernetes/blob/3d5ebcc37f8eef9a3619a094ed12bffb1775eae1/test/e2e/framework/test_context.go#L466-L492) error for DO, so we might need to update [this function](https://github.com/kubernetes/kops/blob/f1fbde95e69c9fda23d8074458eab1a03050eca5/tests/e2e/pkg/tester/tester.go#L157-L162) to not set --provider for DO clusters. We may need to do the same with the other flags in the nearby functions (--gce-zone, --gce-multizone, --gce-region, all defined in e2e.test [here](https://github.com/kubernetes/kubernetes/blob/3d5ebcc37f8eef9a3619a094ed12bffb1775eae1/test/e2e/framework/test_context.go#L344-L347))

Reply:
and feel free to open a test-infra PR now to update the job with these updated kubetest2 commands. We can merge it and test the job against this PR and then proceed with merging this.